lib/matplotlib/quiver.py

```diff
@@ -779,142 +779,141 @@ def _h_arrows(self, length):
 
 
 _barbs_doc = r"""
-Plot a 2-D field of barbs.
+Plot a 2D field of barbs.
 
-Call signatures::
+Call signature::
+
+  barbs([X, Y], U, V, [C], **kw)
```

Comment:
Does this notation un-ambiguously specify that three args means `U, V, C` and four args means `X, Y, U, V`?

Reply:
I would say so: Every square bracket is optional.

The resulting three and four argument forms make it unambiguous. This is not much different than https://matplotlib.org/devdocs/api/_as_gen/matplotlib.axes.Axes.pcolormesh.html allowing only one or three arguments.